hipogram/posts/views.py

```diff
@@ -71,27 +71,27 @@ class PostUpdateView(UpdateView):
     pk_url_kwarg = 'post_id'
 
 
-@login_required()
-def like_post(request, post_id):
-    like, created = Like.objects.get_or_create(user=request.user, post_id=post_id)
+class LikeView(LoginRequiredMixin, View):
 
-    if not created:
-        like.delete()
+    def get(self, request, *args, **kwargs):
+        post_id = request.GET.get("post_id")
+        like, created = Like.objects.get_or_create(user=request.user, post_id=post_id)
 
-    return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+        if not created:
+            like.delete()
 
+        return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
 
-@login_required()
-def rate_post(request, post_id):
 
-    if request.method == 'POST':
-        form = RatePostForm(request.POST)
+class RateView(LoginRequiredMixin, View):
 
-    # if not form.is_valid():
-        # do something
+    def get(self, request, *args, **kwargs):
 
-        rate, _ = Rate.objects.get_or_create(user=request.user, post_id=post_id)
-        rate.value = form.data['value']
-        rate.save()
+        if request.method == 'POST':
```

Comment:
You need to override `post` method if you want to handle `POST` requests.

Reply:
Makes sense, working on it.